src/Statement.py

```diff
@@ -36,6 +37,15 @@ def get_category(self):
             category = "Entnahme"
         elif self.is_fee(booking_type):
             category = "Gebühren"
+        elif self._config.get_special_entry_regex() and self._config.get_special_entry_regex().match(booking_type):
+            if value > 0:
+                category = "Zinsen"
+            elif value < 0:
+                category = "Gebühren"
+            else:
+                logging.debug(self._statement)
+        elif self._config.get_ignorable_entry_regex() and self._config.get_ignorable_entry_regex().match(booking_type):
+            category = "Ignored"
```

Comment:
Maybe move the if statements to its own methods (see is_fee). This is mostly to reduce the complexity of the overall function.
This could look something like:
```suggestion
        elif self.is_special_entry_interest(booking_type):
            category = "Zinsen"
        elif self.is_special_entry_fee(booking_type):
            category = "Gebühren"
        elif self.is_ignorable(booking_type):
            category = "Ignored"

    def is_ignorable(self, booking_type):
        return self._config.get_ignorable_entry_regex() and self._config.get_ignorable_entry_regex().match(booking_type)

    def is_special_entry_interest(self, booking_type):
        return self._config.get_special_entry_regex() and self._config.get_special_entry_regex().match(booking_type) and self.value() >= 0

    def is_special_entry_fee(self, booking_type):
            return self._config.get_special_entry_regex() and self._config.get_special_entry_regex().match(booking_type) and self.value() < 0
```

Overall (in retrospect) not too happy with how much this if branching explodes :) I should have come up with a better way to handle these different categories.

Reply:
Thanks, I tried to make it a little more clear in the refactored version, let me know what you think :)

Reply:
I also put the code for handling the special mintos case in a separate function, I am not too happy about the whole part, but the format of the csv does not make it easy to parse the category :|